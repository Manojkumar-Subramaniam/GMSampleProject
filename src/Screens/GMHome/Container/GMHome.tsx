import React, { useEffect } from 'react'
import { View, FlatList, SafeAreaView, Image, Text, TouchableOpacity } from 'react-native'
import CommitSummary from '../Component/CommitSummary'
import styles from '../Styles/GMHomeStyles'
import gitCommitApiResponseModal from '../../../Modals/gitCommitModal'
import { useNavigation } from '@react-navigation/native';
import colors from '../../../Utils/colors'
import fonts from '../../../Utils/fonts'
import { Strings } from '../../../Utils'
import fetchApi from '../../../Network/fetchApi'



const renderCommitItem = (item: gitCommitApiResponseModal, navigation: any) => {

    return (
        <CommitSummary
            data={item}
            onpress={() => navigation.navigate('CommitInfo', { data: item })} />
    )
}

const GMHome: React.FC = () => {
    const navigation = useNavigation()

    const [isDataLoaded, setLoadComplete] = React.useState<boolean>(false)
    const [commitData, setCommitData] = React.useState<gitCommitApiResponseModal[]>([])

    React.useEffect(() => {
        const fetchdata = async () => {
            let data = await fetchApi<gitCommitApiResponseModal[]>('rockimanj', 'GMSampleProject')
            setLoadComplete(true)
            setCommitData(data)
        }
        fetchdata()
    }, [isDataLoaded])

    return (
        <SafeAreaView style={!isDataLoaded ? { backgroundColor: colors.BLACK, flex: 1 } : {}}>
            {!isDataLoaded ? <Image resizeMode='center' source={require('../../../Assets/GM.gif')} style={styles.loader} /> :
                <View style={styles.rootView}>
                    {commitData.length > 0 ?
                        <FlatList
                            data={commitData}
                            renderItem={({ item }) => renderCommitItem(item, navigation)}
                            inverted={false}
                            showsVerticalScrollIndicator={false}
                            ItemSeparatorComponent={() => <View style={{ height: 24 }} />}
                        /> : <TouchableOpacity style={styles.retryBtn} onPress={() => { setLoadComplete(false) }}>
                            <Text style={[styles.retryTxt, fonts.fontNormalWhite]}>
                                {Strings.RETRY_TXT}
                            </Text>
                        </TouchableOpacity>}
                </View>}
        </SafeAreaView>
    )
}

export default GMHome