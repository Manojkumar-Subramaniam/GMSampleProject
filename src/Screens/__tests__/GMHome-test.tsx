import React from 'react';
import GMHome from '../GMHome/Container/GMHome';
import Enzyme, { shallow } from 'enzyme';
import mockApi from '../../../mockStore';
import { shallowToJson } from 'enzyme-to-json';
import { useNavigation } from '@react-navigation/native';
jest.mock('@react-navigation/native');

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

describe('GM Home Snapshop', () => {
  useNavigation.mockReturnValue({
    navigate: jest.fn(),
  });
  React.useState = jest.fn();

  it('GM Home Loader', () => {
    let setLoadComplete = jest.fn();
    setLoadComplete.mockClear();
    React.useState = () => [false, setLoadComplete];

    let setCommitData = jest.fn();
    setCommitData.mockClear();
    React.useState = () => [[], setCommitData];

    const wrapper = shallow(<GMHome />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('GM Home with data', () => {
    let setLoadComplete = jest.fn();
    setLoadComplete.mockClear();
    React.useState = () => [true, setLoadComplete];

    let setCommitData = jest.fn();
    setCommitData.mockClear();
    React.useState = () => [mockApi, setCommitData];
    const wrapper = shallow(<GMHome />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('GM Home on failure API', () => {
    let setLoadComplete = jest.fn();
    setLoadComplete.mockClear();
    React.useState = () => [true, setLoadComplete];

    let setCommitData = jest.fn();
    setCommitData.mockClear();
    React.useState = () => [[], setCommitData];
    const wrapper = shallow(<GMHome />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
