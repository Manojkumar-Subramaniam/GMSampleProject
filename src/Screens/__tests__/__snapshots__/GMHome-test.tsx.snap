// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GM Home Snapshop GM Home Loader 1`] = `
<ForwardRef(SafeAreaView)
  style={Object {}}
>
  <View
    style={
      Object {
        "paddingHorizontal": 16,
        "paddingTop": 30,
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#34a8eb",
          "borderRadius": 20,
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFff",
              "margin": 16,
            },
            Object {
              "color": "#FFFFff",
              "fontSize": 16,
              "fontWeight": "normal",
            },
          ]
        }
      >
        No Data Found. Tap to retry
      </Text>
    </TouchableOpacity>
  </View>
</ForwardRef(SafeAreaView)>
`;

exports[`GM Home Snapshop GM Home on failure API 1`] = `
<ForwardRef(SafeAreaView)
  style={Object {}}
>
  <View
    style={
      Object {
        "paddingHorizontal": 16,
        "paddingTop": 30,
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#34a8eb",
          "borderRadius": 20,
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFff",
              "margin": 16,
            },
            Object {
              "color": "#FFFFff",
              "fontSize": 16,
              "fontWeight": "normal",
            },
          ]
        }
      >
        No Data Found. Tap to retry
      </Text>
    </TouchableOpacity>
  </View>
</ForwardRef(SafeAreaView)>
`;

exports[`GM Home Snapshop GM Home with data 1`] = `
<ForwardRef(SafeAreaView)
  style={Object {}}
>
  <View
    style={
      Object {
        "paddingHorizontal": 16,
        "paddingTop": 30,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "commit": Object {
              "author": Object {
                "date": "2022-03-31T21:53:32Z",
                "email": "manojkumarsubramaniam@Manojkumars-MacBook-Pro.local",
                "name": "Manojkumar Subramaniam",
              },
              "message": "created RN Sample project",
            },
            "sha": "e29a7e3074e477d9567aa0953f5c6171ba805e1c",
          },
          Object {
            "commit": Object {
              "author": Object {
                "date": "2022-03-31T21:53:32Z",
                "email": "manojkumarsubramaniam@Manojkumars-MacBook-Pro.local",
                "name": "Manojkumar Subramaniam",
              },
              "message": " 1created RN Sample project",
            },
            "sha": "e29a7e3074e477d9567aa0953f5c6171ba805e1c",
          },
        ]
      }
      inverted={false}
      renderItem={[Function]}
      showsVerticalScrollIndicator={false}
    />
  </View>
</ForwardRef(SafeAreaView)>
`;
